instrumentation/yesod/src/OpenTelemetry/Instrumentation/Yesod.hs

```diff
@@ -210,20 +217,31 @@ openTelemetryYesodMiddleware
 openTelemetryYesodMiddleware rr m = do
   req <- waiRequest
   mr <- getCurrentRoute
+  semConvStabilityOptIn <- liftIO getSemConvStabilityOptIn
   let mspan = requestContext req >>= Context.lookupSpan
-      sharedAttributes = H.fromList $
-        ("http.framework", toAttribute ("yesod" :: Text))
-          : catMaybes
-            [ do
-                r <- mr
-                pure ("http.route", toAttribute $ pathRender rr r)
-            , do
-                r <- mr
-                pure ("http.handler", toAttribute $ nameRender rr r)
-            , do
-                ff <- lookup "X-Forwarded-For" $ requestHeaders req
-                pure ("http.client_ip", toAttribute $ T.decodeUtf8 ff)
-            ]
+      sharedAttributes =
+        H.fromList $
+          ("http.framework", toAttribute ("yesod" :: Text))
+            : catMaybes
+              [ do
+                  r <- mr
+                  pure ("http.route", toAttribute $ pathRender rr r)
+              , do
+                  r <- mr
+                  pure ("http.handler", toAttribute $ nameRender rr r)
+              , do
+                  ff <- lookup "X-Forwarded-For" $ requestHeaders req
+                  case semConvStabilityOptIn of
+                    Stable -> pure ("client.address", toAttribute $ T.decodeUtf8 ff)
+                    Both -> pure ("client.address", toAttribute $ T.decodeUtf8 ff)
```

Comment:
imo it's clearer to use `Just` here. More concrete so it's more obvious what it's doing (vs what monad am I in for `pure`?) and contrasts better with Nothing on the next line

V minor though don't think you need to change this

Reply:
Changed it.